app/models/certification.rb

```diff
@@ -1,3 +1,10 @@
 class Certification < ActiveRecord::Base
   belongs_to :user
+  belongs_to :training_session
+
+  validates :user, presence: true
+  validates :trainer_id, presence: true
```

Comment:
This might not be necessary. The certification comes from a training_session which has a trainer already.